### PR TITLE
install.sh: Add remote install support

### PR DESCRIPTION
**Pull request by nhooyr**
Closes cdr/sshcode#185, #1996, and pretty much archives that repository.

See also #1991

### CODE REVIEW

**Comment by nhooyr** [comment]
Yea I think I'ma update all the docs/messages when installing remotely and modify fetch to fetch locally and then copy over ssh so that it works in closed off environments.

File: install.sh [hunk 1]
```diff
@@ -17,27 +17,44 @@ usage() {
...
+
+Pass in user@host to install code-server on user@host over ssh.
+Pass --start to forward the code-server port so that you can immediately access it.
+
+If you rerun the script, code-server will be updated only as required.
 ${not_curl_usage-}
 Usage:
 
-  $arg0 [--dry-run] [--version X.X.X] [--method detect] [--prefix ~/.local]
+  $arg0 [--dry-run] [--version X.X.X] [--method detect] [--prefix ~/.local] [--start] [user@host]
```
Comment (kylecarbs) [inline install.sh:31]: In `--method detect` the `detect` keyword is variable.

In `--start` we have `user@host` separated. Is this intentional?

Comment (nhooyr) [reply install.sh:31]: Yes, `--start` can be used without `user@host`.

Comment (kylecarbs) [reply install.sh:31]: Ah I understand.

File: install.sh [hunk 2]
```diff
@@ -446,18 +470,36 @@ arch() {
...
+  fi
   if [ ! "${DRY_RUN-}" ]; then
+    sh_f "$@"
+  fi
+}
+
+# Always runs.
+sh_f() {
+  if [ "$SSH_ARGS" ]; then
+    mkdir -p ~/.ssh/sockets
```
Comment (SuperSandro2000) [inline install.sh:490]: The ``~/.ssh`` directory requires to have 700 for openssh to accept it. It should probably be chmoded to that.

File: install.sh [hunk 3]
```diff
@@ -494,10 +513,10 @@ echoerr() {
   echoh "$@" >&2
 }
 
-# humanpath replaces all occurances of " $HOME" with " ~"
-# and all occurances of '"$HOME' with the literal '"$HOME'.
+# humanpath replaces all occurances of " $RHOME" with " ~"
+# and all occurances of '"$RHOME' with the literal '"$RHOME'.
 humanpath() {
-  sed "s# $HOME# ~#g; s#\"$HOME#\"\$HOME#g"
+  sed "s# $RHOME# ~#g; s#\"$RHOME#\"\$RHOME#g"
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply install.sh:542]: ``~`` is not supported in early init scripts before the shell started. ``$HOME`` is preferred.

File: install.sh (hunk 2 above)
Comment (nhooyr) [reply install.sh:490]: Good catch!

**Comment by nhooyr** [comment]
This got messy quickly, opening a follow up PR without `--start` and no support for hosts without internet which I think is fine for now.

**Pull request closed without merging**